pkg/app/accounts/usernameonly.go

```diff
@@ -38,8 +38,12 @@ func NewUsernameOnlyAccountManager() *UsernameOnlyAccountManager {
 	return &UsernameOnlyAccountManager{}
 }
 
-func (m *UsernameOnlyAccountManager) UserFromRequest(r *http.Request) (User, error) {
-	return userFromRequest(r)
+func (m *UsernameOnlyAccountManager) UserFromRequest(w http.ResponseWriter, r *http.Request) (User, error) {
+	user, err := userFromRequest(r)
+	if user == nil {
+		http.Redirect(w, r, "/username", http.StatusSeeOther)
+	}
+	return user, err
```

Comment:
Out of curiosity, is `err` still valid though redirection to `/username` brings the cookie, in case of CO WebUI?

Reply:
I think you are correct. `err` should be no longer valid after successfully logging in by redirecting to `/username`. I removed the error code response in my new patch set.

Reply:
Resolving this. I think this thread is now outdated.